src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -454,6 +455,8 @@ public AzureVMAgentTemplate(
 
         // Forms data which is not persisted
         labelDataSet = Label.parse(labels);
+
+        this.tags = new ArrayList<>();
```

Comment:
I don't think you want this, won't it clear the list on every save. You can set it on the field to default it or even just do it in the getter to be safe